src/pages/index.js

```diff
@@ -92,26 +149,98 @@ function fillProfileForm() {
 
 profileEditButton.addEventListener("click", () => {
   formValidators["profile-form"].resetValidation();
+  popupWithForms["edit-modal"].resetButtonText();
   popupWithForms["edit-modal"].open();
   fillProfileForm();
 });
 
+profileAvatar.addEventListener("mouseover", () => {
+  avatarEditButton.classList.add("profile__avatar-button_visible");
+});
+
+avatarEditButton.addEventListener("click", () => {
+  //reset the validation before opening it
+  formValidators["avatar-form"].resetValidation();
+  popupWithForms["avatar-modal"].resetButtonText();
+  popupWithForms["avatar-modal"].open();
+});
+
+const updateProfilePicture = (inputValues) => {
+  api
+    .updateProfilePicture(inputValues["avatar-link"])
+    .then((res) => {
+      profileInfo.updateUserAvatar(res["avatar"]);
+      popupWithForms["avatar-modal"].changeButtonText();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  //retrieveProfileInfo();
+  /*api
+    .loadUserInfo()
+    .then((profileData) => {
+      popupWithForms["avatar-modal"].changeButtonText();
+      profileInfo.updateUserAvatar(profileData["avatar"]);
+    })
+    .catch((err) => {
+      console.error(err);
+    });*/
+  popupWithForms["avatar-modal"].close();
+};
+
 function updateProfileModal(inputValues) {
   profileInfo.setUserInfo(inputValues["Name"], inputValues["About me"]);
+  api
+    .editUserInfo(inputValues["Name"], inputValues["About me"])
+    .then((res) => {
+      popupWithForms["edit-modal"].changeButtonText();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
   popupWithForms["edit-modal"].close();
 }
 
 const generateCard = (cardData) => {
-  const card = new Card(cardData, cardTemplateSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardTemplateSelector,
+    handleImageClick,
+    (card) => {
+      confirmationPopups["delete-card-modal"].open();
+      confirmationPopups["delete-card-modal"].setHandleFormSubmit(() => {
+        api
+          .deleteCard(card.getId())
+          .then(() => {
+            card.deleteCard();
+            confirmationPopups["delete-card-modal"].close();
+          })
+          .catch((err) => {
+            console.error(err);
+          });
+      });
+    },
+    handleCardLike
+  );
+  cardsOnPage[cardData.name] = card;
   return card.generateCard();
 };
 
 const cardsList = new Section(
   {
-    items: initialCards,
-    renderer: (cardItem) => {
-      const cardElement = generateCard(cardItem);
-      cardsList.addItem(cardElement);
+    renderer: () => {
+      api
+        .checkStatus(api.loadUserInfo(), api.getInitialCards())
```

Comment:
Please, delete it from `Section`.  You don't need to call `api.loadUserInfo()` when you render cards (you don't use this info)


And `renderer` is for rendering an array of cards rather than making requests to the server